nixos/modules/services/networking/cjdns.nix

```diff
@@ -91,6 +89,16 @@ in
         '';
       };
 
+      extraConfig = mkOption {
```

Comment:
My recommendation is to have a generic `config` option for the *whole* config. The type of it should be representative of json (I'll help with that if you have trouble) (ideally you should add this type in `lib/types.nix` for future uses). Then instead of setting the opaque options above (line 47), we can use
```
{
  options = ...;
  config = {
    services.cjdns.config = {
      admin = {
        bind = cfg.admin.bind;
        password = "@CJDNS_ADMIN_PASSWORD@";
      };
    };
  };
}
```

This has the benefits of:
- Allowing users to overwrite default config values with `mkDefault` and `mkForce`.
- Having proper merging behaviour (in the current implementation, values set in `extraConfig` would silently overwrite fixed options). Use of `mkDefault` is recommended in this implementation for some settings.
- Allows the user to view the full config by evaluating this option: `nix-instantiate --eval '<nixpkgs/nixos>' -A config.services.cjdns.config --strict`

Reply:
I support that recommendation, but why not file it as its own feature request, since it can apply to many other modules? I fail to understand why it's crucial for this one.

Reply:
Yeah it's not crucial, I guess I'm asking for a bit too much.